idx/memory/tag_query_test.go

```diff
@@ -559,6 +559,108 @@ func TestExpressionParsing(t *testing.T) {
 	}
 }
 
+func TestSpecialCharactersInTags(t *testing.T) {
+	type testCase struct {
+		id         schema.MKey
+		lastUpdate int64
+		tags       []string
+	}
+
+	ids = []schema.MKey{
+		test.GetMKey(0),
+		test.GetMKey(1),
+		test.GetMKey(2),
+		test.GetMKey(3),
+		test.GetMKey(4),
+		test.GetMKey(5),
+		test.GetMKey(6),
+	}
+
+	data := []testCase{}
+
+	for i := 0; i < 6; i++ {
+		data = append(data, testCase{ids[i], 1, []string{}})
+	}
+
+	blackList := []int{
+		59,  // ; not allowed in values
+		126, // ~ can't query values starting with ~ because =~ looks like a regular expression query
```

Comment:
can't you just specify the actual characters like `':'`

Reply:
yes, that works too: https://github.com/grafana/metrictank/pull/1217/commits/42c75fae37ed4c2191d762cec238397ed7e4e2df